pkg/pathutils/pathutils.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package pathutils
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+)
+
+// https://github.com/bazelbuild/bazel/blob/8346ea4c/src/main/cpp/workspace_layout.cc#L37
+var WorkspaceFilenames = []string{"WORKSPACE", "WORKSPACE.bazel"}
+
+// CobraRunEFn is the function signature for the cobra RunE function in the cobra.Command.
+type CobraRunEFn func(cmd *cobra.Command, args []string) (exitErr error)
+
+// RunWorkspaceFn is the function signature based on CobraRunEFn that includes the workspace root.
+type RunWorkspaceFn func(workspaceRoot string, cmd *cobra.Command, args []string) (exitErr error)
+
+// InvokeCmdInsideWorkspace verifies if the current working directory is inside a Bazel workspace,
+// then invokes the provided toRun function, injecting the found workspace root path.
+func InvokeCmdInsideWorkspace(toRun RunWorkspaceFn) CobraRunEFn {
+	wd, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	return invokeCmdInsideWorkspace(defaultWorkspaceFinder, wd, toRun)
+}
+
+func invokeCmdInsideWorkspace(
+	finder Finder,
+	wd string,
+	toRun RunWorkspaceFn,
+) CobraRunEFn {
+	return func(cmd *cobra.Command, args []string) (exitErr error) {
+		workspacePath, err := finder.Find(wd)
+		if err != nil {
+			return fmt.Errorf("failed to run command %q: %w", cmd.Use, err)
+		}
+		if workspacePath == "" {
+			err = fmt.Errorf("the current working directory %q is not a Bazel workspace", wd)
```

Comment:
note for later (maybe a TODO?) we should offer to help if the user actually meant the cwd to be a workspace - go interactive and into a setup mode

Reply:
Created https://github.com/aspect-build/aspect-cli/issues/85.